### ISSUE TITLE
Telemetry: Collect enabled core modules

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#5185]
This is part of the #285 meta.

**Comment by stpaultim** [comment]
I agree that this could be some very useful data.

**Comment by BWPanda** [comment]
PR: https://github.com/backdrop/backdrop/pull/4107

![image](https://user-images.githubusercontent.com/2385329/175756524-3a30a961-9179-485c-9189-81ede57cee19.png)

Might have to start thinking about how to format the data differently on the Telemetry page compared to how it's sent to B.org...

**Comment by indigoxela** [comment]
TBH, I'm not so sure if this provides any useful info. A very long string imploded with "," will almost always be unique per site. And server-side no arrays are supported AFAIK, so it probably has to be that pointless long string.

**Comment by stpaultim** [comment]
Knowing which core modules are enabled is helpful. BUT, having the data in a single long string, might not be helpful. What I think MIGHT be helpful is a list of all core modules with a note next to each one indicating what percentage of reporting sites have that module enabled.

**Comment by jenlampton** [comment]
Is this a bug fix or a new feature? I think it will need tests, so changing to new feature.

**Comment by klonos** [comment]
>A very long string imploded with "," will almost always be unique per site. And server-side no arrays are supported AFAIK, so it probably has to be that pointless long string.

Yeah 😓 ...I don't see an easy way to be getting metrics about all enabled modules. Not saying that we should drop this idea, but perhaps for the time being it would be easier if we collected one metric per module. ...not ideal, I know, but otherwise we need some serious rework of the code on the receiving end on b.org 🤔

**Comment by klonos** [comment]
...here's a though: if you exclude all the test modules and all the hidden/required modules, core currently has 8 disabled vs 37 enabled modules in the standard installation profile (which is the profile being used by 98% of the sites according to https://backdropcms.org/project/backdrop/telemetry). Although people might have disabled some of the enabled-by-default OOTB modules post installation, the disabled-by-default modules are the following:

book
contact
entityreference
language
locale
simpletest
syslog
translation

Out of that list, we only have issues in the queue for entityreference (to have it be enabled by default - see #5913) and for syslog (to have it be removed from core altogether - see #1260). I don't think that we currently have any intention to remove simpletest, nor book, nor contact, nor any of the language/locale/translation modules either. So how about for starters we gather some "targeted" metrics for syslog and entityreference specifically, and when we have a better way to gather array-based data, we adjust telemetry to gather metrics for all enabled and disabled core modules?

**Comment by klonos** [comment]
We have discussed this during the dev meeting today, and have concluded that we'd better collect metrics for disabled core modules instead. Also, due to the limitation of the current implementation of telemetry, we unfortunately will need to collect one metric per module instead of an array of modules (otherwise we'd be getting combinations of modules as results in the telemetry, which is not ideal).

I'll work on a PR for this.

**Comment by klonos** [comment]
...I've worked on this on my local, and this is what I have so far:
![image](https://github.com/backdrop/backdrop-issues/assets/2423362/f6d22e34-5a8f-4c6e-a274-5ba997c01500)

The code generates a list of all disabled core modules on the site, and sends an individual metric for each one of them. I realized though that this is not ideal, nor what we want: the above will result in individual metrics in https://backdropcms.org/project/backdrop/telemetry, which will be showing various disabled modules, which is good 👍🏼 ...however, each metric will be showing 100% as its percentage, since all sites will be reporting only disabled modules ...which means that the only useful metric will be the count (and then we'd need to be comparing that to the other non-disabled-modules metrics to get a total and do the maths in our heads ...no good 👎🏼).

I think that as things are at the moment (with the limitations of the current implementation of our Telemetry system), we'd be better off if we hard-coded the list of modules that we wanted metrics for, and we gathered metrics on whether these specific modules are enabled or disabled (as opposed to sending metrics of ALL disabled modules on the various sites).

Thoughts?

**Comment by klonos** [comment]
...OK, I filed a PR that checks a specific, hard-coded list of core modules, end reports whether these are enabled or disabled: https://github.com/backdrop/backdrop/pull/4444

![image](https://github.com/backdrop/backdrop-issues/assets/2423362/2d11a692-8c21-4aae-b275-3b174b93fa96)
